File: Python/binarySearch.py
n=(int)(input("Enter the size of the array"))
arr=[]
print("Enter the array elements in sorted order")
for i in range(n):
    arr.append((int)(input()))
s=(int)(input("Enter the search element"))
start,end,c=0,n-1,0
while start<=end:
    mid=(start+end)//2
    if arr[mid]==s:
        print("Element found at ",mid+1," position")
        c=c+1
        break
    elif arr[mid]<s:
        end=mid-1
    else:
        start=mid+1
if c==0:
    print("Element not found")

File: Searching/binarySearch.c
//Program: Binary Search
//Language: C
//This algorithm only works on sorted array

#include<stdio.h>
void main()
{
    int start=0,end,mid,s,i,n,c=0;
    printf("Enter the size of the array\n");
    scanf("%d",&n);

    int a[n];
    printf("Enter the sorted array elements\n");
    for(i=0;i<n;i++)
        scanf("%d",&a[i]);
    
    printf("Enter the search element\n");
    scanf("%d",&s);

    end=n-1;
    while (start<=end)
    {
        mid=(start+end)/2;
        if(a[mid]==s)
            {
                printf("Element found at %d position \n",mid+1);
                c++;
                break;
            }
        else if(a[mid]<s)
            start=mid+1;
        else
            end=mid-1;
    }

    if(c==0)
        printf("Element not found\n");   

}